lua/entities/gmod_wire_expression2/core/http_cl.lua

```diff
@@ -0,0 +1,157 @@
+--[[
+
+	Clientside HTTP library
+	 - swadical 🍌
+
+]]
+
+--	Clientside HTTP is separate from serverside HTTP:
+--		Serverside HTTP allows greater speeds than clientside HTTP and it is
+--		up to the server owner to decide which extension should be enabled
+--		through a convar (found in core/http.lua)
+
+--	This library will mirror the functionality of core/http.lua as best as
+--	possible, making sure any existing e2s which make use of the http library
+--	are not broken by this extension
+
+-- should we use clientside HTTP requests?
+local cvar_useClient = CreateConVar("wire_expression2_http_client","1",FCVAR_ARCHIVE)
+
+E2Lib.clHTTP = {}
+local lib = E2Lib.clHTTP
+
+lib.netMsgID = "wire_expression2_cl_http" -- open to change
+lib.currentUID = 0
+
+-- number function(void)
+function lib:newUID()
+	self.currentUID = (self.currentUID + 1) % 255
+	return self.currentUID -- return a number between 0 and 254
+end
+
+lib.rawRequest = http.Fetch
+lib.segmentSize = 4 * 1024-- 4 kilobytes
+lib.segmentBits = math.min(math.ceil(math.log(lib.segmentSize,2)) + 2,32)
+lib.sendInterval = 0.5-- seconds
+lib.HTTP_REQUEST_FAILED = 1024--enum for HTTP failure
+
+-- void function(string id, number interval, function func)
+-- Internal: Do not call.
+function lib.timerCoroutine(id,interval,func)
+	func = coroutine.wrap(func)
```

Comment:
If this were more generic (ie. used `coroutine.create()` and `coroutine.status()` to run it until it's done, instead of relying on an explicit return status from the wrapped function) then it'd be good to have it in `wireshared.lua` - there's lots of places where this would be a useful construct.